pkg/api/plugins.go

```diff
@@ -317,3 +380,13 @@ func translatePluginRequestErrorToAPIError(err error) response.Response {
 
 	return response.Error(500, "Plugin request failed", err)
 }
+
+func shouldExclude(fi os.FileInfo) bool {
+	normalizedFilename := strings.ToLower(fi.Name())
+
+	isUnixExecutable := fi.Mode()&0111 == 0111
+	isWindowsExecutable := strings.HasSuffix(normalizedFilename, ".exe")
```

Comment:
I'm quite curious.. since this has broken so many folks' installations of various plugins.. what problem does this change solve?

Reply:
The intention was to tidy up the API and allow only necessary plugins files to be served (and primarily keep things consistent with how it behaved with our previous middleware setup). The only files that should be executable in a plugin are the backend binaries, but this has highlighted unforeseen scenarios in people's setups where this wasn't the case. In most reported cases it relates to an issue with [Windows for Docker](https://docs.docker.com/desktop/faqs/#can-i-change-permissions-on-shared-volumes-for-container-specific-deployment-requirements) and others who are running Synology NAS

Reply:
Than you very much for the explanation!